net/tshttpproxy/tshttpproxy_synology.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package tshttpproxy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"tailscale.com/util/lineread"
+	"tailscale.com/version/distro"
+)
+
+func init() {
+	sysProxyFromEnv = linuxSysProxyFromEnv
+}
+
+var (
+	distroOnce         sync.Once
+	distroProxyFromEnv func(*http.Request) (*url.URL, error)
+)
```

Comment:
this is all overkill. distro.Get() is already using a sync.Once.

The it's fine to do a string comparison per call to `linuxSysProxyFromEnv`.

```go
func linuxSysProxyFromEnv(req *http.Request) (*url.URL, error) {
    if distro.Get() == distro.Synology {
         return synologyProxyFromConfigCached(req)
    }
    return nil, nil
}
```

Reply:
Fair point, the once was also guarding the update, I didn't look at Get to see that it was once'd, makes sense.